cli/tools/doc.rs

```diff
@@ -109,49 +135,101 @@ pub async fn print_docs(
         },
       )?;
 
-      let mut doc_nodes = vec![];
+      let mut doc_nodes_by_url =
+        IndexMap::with_capacity(module_specifiers.len());
 
-      for module_specifier in module_specifiers {
-        let nodes = doc_parser.parse_with_reexports(&module_specifier)?;
-        doc_nodes.extend_from_slice(&nodes);
+      for module_specifier in &module_specifiers {
+        let nodes = doc_parser.parse_with_reexports(module_specifier)?;
+        doc_nodes_by_url.insert(module_specifier.clone(), nodes);
       }
 
       if doc_flags.lint {
         let diagnostics = doc_parser.take_diagnostics();
         check_diagnostics(&diagnostics)?;
       }
 
-      doc_nodes
+      doc_nodes_by_url
     }
   };
 
-  if doc_flags.json {
-    write_json_to_stdout(&doc_nodes)
+  if let Some(html_options) = doc_flags.html {
+    generate_docs_directory(&doc_nodes_by_url, html_options)
+      .boxed_local()
+      .await
   } else {
-    doc_nodes.retain(|doc_node| doc_node.kind != doc::DocNodeKind::Import);
-    let details = if let Some(filter) = doc_flags.filter {
-      let nodes =
-        doc::find_nodes_by_name_recursively(doc_nodes, filter.clone());
-      if nodes.is_empty() {
-        bail!("Node {} was not found!", filter);
-      }
-      format!(
-        "{}",
-        doc::DocPrinter::new(&nodes, colors::use_color(), doc_flags.private)
-      )
-    } else {
-      format!(
-        "{}",
-        doc::DocPrinter::new(
-          &doc_nodes,
-          colors::use_color(),
-          doc_flags.private
-        )
-      )
-    };
+    let doc_nodes: Vec<doc::DocNode> =
+      doc_nodes_by_url.values().flatten().cloned().collect();
+    print_docs(doc_flags, doc_nodes)
+  }
+}
+
+async fn generate_docs_directory(
+  doc_nodes_by_url: &IndexMap<ModuleSpecifier, Vec<doc::DocNode>>,
+  html_options: DocHtmlFlag,
+) -> Result<(), AnyError> {
+  let cwd = std::env::current_dir().context("Failed to get CWD")?;
+  let output_dir_resolved = cwd.join(&html_options.output);
+
+  let options = deno_doc::html::GenerateOptions {
+    package_name: html_options.name,
+  };
 
-    write_to_stdout_ignore_sigpipe(details.as_bytes()).map_err(AnyError::from)
+  let files = deno_doc::html::generate(options, doc_nodes_by_url)
+    .context("Failed to generate HTML documentation")?;
+
+  let path = &output_dir_resolved;
+  let _ = std::fs::remove_dir_all(path);
+  std::fs::create_dir(path)
+    .with_context(|| format!("Failed to create directory {:?}", path))?;
+
+  let no_of_files = files.len();
+  for (name, content) in files {
+    let this_path = path.join(name);
+    let prefix = this_path.parent().with_context(|| {
+      format!("Failed to get parent path for {:?}", this_path)
+    })?;
+    std::fs::create_dir_all(prefix)
+      .with_context(|| format!("Failed to create directory {:?}", prefix))?;
+    std::fs::write(&this_path, content)
+      .with_context(|| format!("Failed to write file {:?}", this_path))?;
   }
+
+  println!(
```

Comment:
Should this be `log::info!`?

Reply:
Using `log::info!` seems to not print this info by default. I'm gonna revert it